packages/form-core/test-suites/FormatMixin.suite.js

```diff
@@ -401,6 +410,38 @@ export function runFormatMixinSuite(customConfig) {
 
         expect(el.formattedValue).to.equal(`foo: ${generatedModelValue}`);
       });
+
+      it('will block the user from inputting invalid values when using a preprocessor', async () => {
+        const preprocessorSpy = sinon.spy(value => value.replace(/[o]/g, ''));
+        const val = generateValueBasedOnType({ viewValue: true }) || 'init-value';
+
+        // Create a copy and run the same preprocessing on it.
+        // So we can check later that the element preprocessor runs
+        // and gives the same output.
+        let processedVal = val;
+        if (typeof val === 'string') {
+          processedVal = val.replace(/[o]/g, '');
+        }
+
+        const el = /** @type {FormatClass} */ (await fixture(html`
+          <${elem} .preprocessor=${preprocessorSpy}>
+            <input slot="input" value="${val}">
+          </${elem}>
+        `));
+
+        if (cfg.tagString?.startsWith('lion-input-date')) {
+          // For date and datepicker the stringified modelValue will be quite different
+          // from the original view value, due to date parsing etc.
+          // So for such inputs we don't test this preprocessor.
+          return;
+        }
+        expect(preprocessorSpy.callCount).to.equal(1);
+        expect(`${el.modelValue}`).to.equal(`${processedVal}`);
```

Comment:
So we basically want to know whether modelValue did not become `Unparseable`?

It is not something that inherently is true for preprocessors, right? (it could be the outcome though, depending on the parser)

Maybe we should test the viewValue (`.value`) instead?

Like:

```js
let val = generateValueBasedOnType({ viewValue: true }) || 'init-value';
val = val + '-'

const preprocessorSpy = sinon.spy(value => value.replace(/\-$/g, ''));

```